language-server/src/server.js

```diff
@@ -149,9 +178,15 @@ documents.onDidChangeContent(async ({ document }) => {
 const validateSchema = async (document) => {
   const diagnostics = [];
 
+  let settings = await getDocumentSettings(document.uri);
+  if (!settings) {
+    connection.console.log("Settings not available");
+    settings = {};
+  }
```

Comment:
Setting it to `globalSettings` would be the correct way of assigning to `settings`. We'll handle it in `getDocumentSettings` to avoid duplication.

Reply:
@wthrajat Sure, right now there's no difference, but my understanding is that `globalSettings` is initialized to the default configs. Currently that's empty because we only have one config and it doesn't have a default, but that's not always going to be true. At some point I expect we will have more configs and some of them might have defaults.

So, the question is, when `connection.workspace.getConfiguration` returns nothing, is the desired behavior that the settings should be empty, or should it be the default configs from `globalSettings`?

As a corollary to that point, should the `globalSettings` get merged into the result from `connection.workspace.getConfiguration` to fill in default for any potentially missing configs that need defaults?

Reply:
As far as my understanding of `globalSettings`, it should be that The global settings are used when the `workspace/configuration` request is not supported by the client. So, I think when `connection.workspace.getConfiguration`
returns nothing, let `settings` be an empty object but when configuration only is not supported by the client then we use the `globalSettings`. Does that seem fine? @wthrajat @jdesrosiers

Reply:
So, I think `globalSettings` being merged into the result depends on the behavior we expect of `globalSettings` as I mentioned above. But what if the user doesn't want globalSettings to be merged, like what if they just want the settings that they have specified in `settings.json` . I maybe wrong but if this might be the case then I don't think `result` and the potentially missing configs should be merged.

Reply:
Thanks for the thoughtful analysis @diyaayay. It sounds like merging isn't the right thing. However, I disagree with the conclusion that `getConfiguration` returning nothing is the same as an empty object. I think an empty object is the user explicitly saying they want no settings set (which is why merging is probably the wrong choice), but returning nothing is the user saying nothing about what settings they want. If the user says nothing about what settings they want, I think it makes sense that they get the `globalSettings`.

Reply:
resolved with https://github.com/hyperjump-io/json-schema-language-tools/pull/25/commits/53c47a4e381f28f2d8f139c14e890e8711ae0844